lib/registry.py

```diff
@@ -387,17 +388,21 @@ def blob_exists_p(self, digest):
       #    requests.head() does not follow redirects, and it seems like a
       #    weird status, so I worry there is a gotcha I haven’t figured out.
       url = self._url_of("blobs", "sha256:%s" % digest)
-      res = self.request("HEAD", url, {200,401,404})
+      res, m = self.request("HEAD", url, {200,401,404})
       return (res.status_code == 200)
 
    def blob_to_file(self, digest, path, msg):
       "GET the blob with hash digest and save it at path."
       # /v2/library/hello-world/blobs/<layer-hash>
       url = self._url_of("blobs", "sha256:" + digest)
       sw = ch.Progress_Writer(path, msg)
-      self.request("GET", url, out=sw)
+      res, m = self.request("GET", url, out=sw)
       sw.close()
 
+      # Check for integrity of downloaded blob by validating its hash digest
+      if (str(digest) != str(m.hexdigest())):
```

Comment:
I'm pretty sure both of these are already strings? i.e. I think you want:
```suggestion
      if (digest != m.hexdigest()):
```